### PR TITLE
Cdn integration

### LINKED ISSUE #374: CDN integration
https://www.notion.so/stream-wiki/CDN-SDK-Integration-Guide-6a221110a82c4b949e43b3e329694383 
https://www.notion.so/stream-wiki/MIME-Type-Standardization-62ff84bf3c2d4640aa3baef95933610e

### PR DESCRIPTION
**Pull request by szuperaz**
Specs: https://www.notion.so/stream-wiki/Attachment-standardization-1e1b45c9f28546b6b4104d1a67264197#d2031dd1002e4025b911902ed357af5f 

# Calculate optimal image size
- Image resize applied for image attachments (single and gallery) and video thumbnails
- Images displayed inside the image carousel are NOT resized

# Display video thumbnails
- The `thumb_url` returned from video file upload is displayed -> only newly uploaded videos will have the `thumb_url` present as this feature required a change inside the SDK
-  Thumbnails can be turned on/off inside the SDK

CSS: https://github.com/GetStream/stream-chat-css/pull/172

closes #374

### CODE REVIEW

File: projects/stream-chat-angular/src/lib/attachment-list/attachment-list.component.spec.ts [hunk 1]
```diff
@@ -104,11 +104,11 @@ describe('AttachmentListComponent', () => {
     expect(queryAttachments().length).toBe(0);
 
     component.attachments = [
-      { type: 'image', img_url: 'url1' },
-      { type: 'file', asset_url: 'url3' },
+      { type: 'image', img_url: 'http://url1' },
```
Comment (szuperaz) [inline projects/stream-chat-angular/src/lib/attachment-list/attachment-list.component.spec.ts:107]: I don't like to mock every dependency of every component, so I had to update attachment URLs across unit tests to be valid URLs. It caused a lot of changes

File: projects/stream-chat-angular/src/lib/attachment-configuration.service.ts [hunk 2]
```diff
@@ -53,17 +61,27 @@ export class AttachmentConfigurationService<
       );
     }
 
-    const height = {
-      gallery: '', // Set from CSS,
-      single: '300px',
-      carousel: '', // Set from CSS
+    // x2 values for retina displays
+    const sizeResctriction = {
+      gallery: { height: 300, width: 300 },
```
Comment (szuperaz) [inline projects/stream-chat-angular/src/lib/attachment-configuration.service.ts:66]: Max height and width come from CSS rules -> CSS defines 150px, we use 300px to ensure appropriate image quality for retina displays

File: projects/stream-chat-angular/src/lib/attachment-configuration.service.ts [hunk 3]
```diff
@@ -36,6 +40,10 @@ export class AttachmentConfigurationService<
   customScrapedImageAttachmentConfigurationHandler?: (
     a: Attachment<T>
   ) => AttachmentConfigration;
+  /**
+   * You can turn on/off thumbnail generation for video attachments
+   */
+  shouldGenerateVideoThumbnail = true;
```
Comment (szuperaz) [inline projects/stream-chat-angular/src/lib/attachment-configuration.service.ts:46]: Based on https://www.notion.so/stream-wiki/CDN-SDK-Integration-Guide-6a221110a82c4b949e43b3e329694383#24fc21e923f644758f30790822cbb462

File: projects/stream-chat-angular/src/lib/attachment-configuration.service.ts (hunk 2 above)
Comment (szuperaz) [reply projects/stream-chat-angular/src/lib/attachment-configuration.service.ts:66]: This shows a potential problem: if an integrator changes the corresponding CSS rules our resizing logic could produce too big/small images -> we should probably create CSS variables and read these constraints from there?

File: projects/stream-chat-angular/src/lib/attachment-configuration.service.ts [hunk 4]
```diff
@@ -124,4 +149,37 @@ export class AttachmentConfigurationService<
...
+  private addResiziParamsToUrl(
+    sizeResctriction: { width: number | undefined; height: number | undefined },
+    url: URL
+  ) {
+    const urlParams = url.searchParams;
+    const originalHeight = Number(urlParams.get('oh')) || 1;
+    const originalWidth = Number(urlParams.get('ow')) || 1;
+
+    const h = sizeResctriction.height
+      ? Math.round(
```
[the remark this answers is not in the document]
Comment (szuperaz) [reply projects/stream-chat-angular/src/lib/attachment-configuration.service.ts:162]: The CDN only accepts integer values, that's why we do a round at the end

File: projects/stream-chat-angular/src/lib/attachment-configuration.service.ts [hunk 5]
```diff
@@ -53,17 +61,27 @@ export class AttachmentConfigurationService<
       );
     }
 
-    const height = {
-      gallery: '', // Set from CSS,
-      single: '300px',
```
Comment (szuperaz) [inline projects/stream-chat-angular/src/lib/attachment-configuration.service.ts:58]: We can also adjust height based on aspect ratio, I'll implement this

File: projects/stream-chat-angular/src/lib/attachment-configuration.service.ts [hunk 6]
```diff
@@ -53,17 +61,52 @@ export class AttachmentConfigurationService<
...
+        );
+        height = `${sizeRestriction.height}px`;
+      }
+    }
+
+    if (sizeRestriction) {
+      // Apply 2x for retina displays
+      sizeRestriction.height *= 2;
+      sizeRestriction.width *= 2;
+      this.addResiziParamsToUrl(sizeRestriction, url);
```
Comment (petyosi) [inline projects/stream-chat-angular/src/lib/attachment-configuration.service.ts:105]: Is this mehtod name missing `ing`? `addResizingParametersToUrl`?

**Review by petyosi — APPROVED** [review]

File: projects/stream-chat-angular/src/lib/attachment-configuration.service.ts (hunk 6 above)
Comment (szuperaz) [reply projects/stream-chat-angular/src/lib/attachment-configuration.service.ts]: Thanks, done

**Pull request merged**